Trifolia.Web/Scripts/TemplateEdit/TemplateEditViewModel.js

```diff
@@ -24,6 +24,113 @@ var templateEditViewModel = function (templateId, defaults) {
     self.AvailableExtensions = ko.observableArray([]);
     self.SelectedAvailableExtensionId = ko.observable();
 
+    self.duplicateNode = function () {
+        if (!self.CurrentNode() || !self.CurrentNode().Constraint()) {
+            return false;
+        }
+
+        var constraint = self.CurrentNode().Constraint();
+
+        if (!constraint.IsPrimitive()) {
+            for (var x = 0; x < self.Constraints().length; x++) {
+                if (constraint.Context() === self.Constraints()[x].Context() && constraint.Id() !== self.Constraints()[x].Id()) {
+                    return true;
+                }
+            }
+        } else {
+            var siblings = constraint.Parent().Children();
+            for (var x = 0; x < siblings.length; x++) {
```

Comment:
I think you meant _.find() here? I made the change and committed it to the branch.